Y0/y0_euler.py

```diff
@@ -0,0 +1,294 @@
+    """Drive the Y0 example."""
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue)
+
+    logger = logging.getLogger(__name__)
+
+    dim = 3
+    order = 1
+    exittol = .09
+    t_final = 0.1
+    current_cfl = 1.0
+    vel_init = np.zeros(shape=(dim,))
+    vel_inflow = np.zeros(shape=(dim,))
+    orig = np.zeros(shape=(dim,))
+    orig[0] = 0.83
+    orig[2] = 0.001
+    #    vel[0] = 340.0
+    vel_inflow[0] = 100.0  # m/s
+    current_dt = 1e-8
+    current_t = 0
+    eos = IdealSingleGas()
+    bulk_init = Lump(numdim=dim, rho0=1.225, p0=100000.0,
+                     center=orig, velocity=vel_init, rhoamp=0.0)
+    inflow_init = Lump(numdim=dim, rho0=1.225, p0=200000.0,
+                       center=orig, velocity=vel_inflow, rhoamp=0.0)
+
+    casename = "pseudoY0"
+    wall = AdiabaticSlipBoundary()
+    from grudge import sym
+    #    boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
+    boundaries = {sym.DTAG_BOUNDARY("Inflow"): PrescribedBoundary(inflow_init),
+                  sym.DTAG_BOUNDARY("Outflow"): DummyBoundary,
+                  sym.DTAG_BOUNDARY("Wall"): wall}
+    constant_cfl = False
+    nstatus = 10
+    nviz = 10
+    rank = 0
+    checkpoint_t = current_t
+    current_step = 0
+    timestepper = rk4_step
+
+    comm = MPI.COMM_WORLD
+    nproc = comm.Get_size()
+    rank = comm.Get_rank()
+    num_parts = nproc
+
+    from meshmode.distributed import (
+        MPIMeshDistributor,
+        get_partition_by_pymetis,
+    )
+
+    global_nelements = 0
+    local_nelements = 0
+
+    if nproc > 1:
+        mesh_dist = MPIMeshDistributor(comm)
+        if mesh_dist.is_mananger_rank():
+
+            mesh = get_pseudo_y0_mesh()
+            global_nelements = mesh.nelements
+            logging.info(f"Total {dim}d elements: {global_nelements}")
+            logging.info(f"Grid BTAGS: {mesh.boundary_tags}")
+            #            print(f"btags = {mesh.boundary_tags}")
+
+            logging.info("Partitioning grid.")
+            part_per_element = get_partition_by_pymetis(mesh, num_parts)
+            logging.info("Sending grid partitions")
+
+            local_mesh = mesh_dist.send_mesh_parts(mesh, part_per_element, num_parts)
+            del mesh
+            logging.info("Grid partitions sent.")
+
+        else:
+            local_mesh = mesh_dist.receive_mesh_part()
+
+        comm.Barrier()
+
+        if rank == 0:
+            logging.info("Grid distributed.")
+    else:
+        logging.info("Reading grid.")
+        local_mesh = get_pseudo_y0_mesh()
+        global_nelements = local_mesh.nelements
+        logging.info("Done. Reading grid.")
+
+    local_nelements = local_mesh.nelements
+
+    comm.Barrier()
+    if rank == 0:
+        logging.info("Making discretization")
+    discr = EagerDGDiscretization(
+        actx, local_mesh, order=order, mpi_communicator=comm
+    )
+    nodes = thaw(actx, discr.nodes())
+    comm.Barrier()
```

Comment:
What makes these barriers necessary?

Reply:
What barriers? -or-  These are not the barriers you are looking for.  ((( :hand: )))